metrics-exporter-prometheus/src/builder.rs

```diff
@@ -566,6 +574,20 @@ fn basic_auth(username: &str, password: Option<&str>) -> HeaderValue {
     header
 }
 
+enum HttpCli {
+    HttpClient(Client<HttpConnector>),
+    HttpsClient(Client<HttpsConnector<HttpConnector>>),
+}
+
+fn create_gateway_client(endpoint: &Uri) -> HttpCli {
+    if endpoint.scheme() == Some(&Scheme::HTTPS) {
+        let https = HttpsConnector::new();
+        HttpCli::HttpsClient(Client::builder().build::<_, hyper::Body>(https))
+    } else {
+        HttpCli::HttpClient(Client::new())
+    }
+}
+
```

Comment:
We should always use the `HttpsConnector`-based version, regardless of the scheme. [`HttpsConnector::new`](https://docs.rs/hyper-tls/latest/hyper_tls/struct.HttpsConnector.html#method.new) mentions that it will handle HTTP vs HTTPS properly, so we can avoid all this extra code.

Reply:
@tobz  You're right. I've updated the code after double-checked the implementation in hyper-tls.